File: CSHARP/Cap07/c07ex08.cs
// c07ex08.cs

using System;
using static System.Console;

class Program
{
    static int Fatorial(int N)
    {
        if (N == 0)
            return 1;
        else
            return N * Fatorial(N - 1);
    }

    public static void Main(string[] args)
    {
        int Limite;

        Write("Qual fatorial: ");
        Limite = int.Parse(ReadLine());

        WriteLine($"Fatorial = {Fatorial(Limite)}");

        Write("\nTecle <Enter> para encerrar... ");
        ReadLine();
    }
}

File: KOTLIN/Cap07/c07ex08.kt
// c07ex08.kt

fun fatorial(N: Int): Int {
    if (N == 0) {
        return 1
    } else {
        return N * fatorial(N - 1)
    }
}

fun main() {
    var limite: Int

    print("Qual fatorial: ")
    limite = readLine()?.toIntOrNull() ?: 0

    println("Fatorial = ${fatorial(limite)}")

    print("\nTecle <Enter> para encerrar... ")
    readLine()
}
